Full UI

r2go-beep opened the pull request:
The UI is complete except for Rating a driver and rating a passenger.

We are getting errors from the backend 

Error: Cannot add or update a child row: a foreign key constraint fails (`unter`.`passenger_trips`, CONSTRAINT `fk_passengerTrips_passengerID` FOREIGN KEY (`passengerID`) REFERENCES `passenger` (`userID`))
    at PromiseConnection.execute (/Users/hanshummel/Documents/GitHub/unter/server/node_modules/mysql2/promise.js:111:22)
    at DBHelper.query (/Users/hanshummel/Documents/GitHub/unter/server/shared/DB.js:25:46)

r2go-beep committed:
Merge pull request #1 from ShreyeshArangath/main

Resolving Conflicts

r2go-beep committed:
Merge branch 'ShreyeshArangath:main' into main

r2go-beep committed:
Added basic functionality to all pages

ShreyeshArangath commented:
When you are trying to rate a driver/passenger, does the driver/passenger exist in the database? This might be why it is throwing the error. To find out what exists in the database, head over to `data.sql` and see the list of drivers and passengers. Lmk if that fixes it, if not ill look into it a bit more

ShreyeshArangath commented on README-class.md:
```diff
@@ -1,4 +1,28 @@
 # CS-4354 Related Details 
 ## Project Description
+
+# API Calls: 
```
Thank you for doing this!

r2go-beep commented:
I do not yet have functionality for rating a driver/passenger. The error is thrown upon creation of a trip with passenger being "tcook@apple.com" which is inside our data upon seeding.
-- Passengers 
INSERT INTO PASSENGER VALUES ("Jack", "Dorsey", '2000-01-21', "jack.dorsey@tesla.com");
INSERT INTO PASSENGER VALUES ("Timothy", "Cook", '1964-01-21', "tcook@apple.com");
INSERT INTO PASSENGER VALUES ("Mark", "Zuckerberg", '1992-01-21', "zucky@meta.com");

r2go-beep committed:
Fixed getting Trip by ID to return 1 object

Updated pages to only pass the trip ID instead of the whole trip.

Updated backend Driver component to only return a trip with matching ID instead of all trips with the same state.
